cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/TimePartitionedFileSetDatasetAvroSource.java

```diff
@@ -83,11 +88,21 @@ protected void addFileSetProperties(FileSetProperties.Builder properties) {
   }
 
   @Override
-  public void prepareRun(BatchSourceContext context) {
-    setInput(context);
-    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(tpfsAvroConfig.schema);
-    Job job = context.getHadoopJob();
-    AvroJob.setInputKeySchema(job, avroSchema);
+  protected void addInputFormatConfiguration(Map<String, String> config) {
+    try {
+      Job job = Job.getInstance();
+      Configuration hConf = job.getConfiguration();
+      hConf.clear();
+
+      Schema avroSchema = new Schema.Parser().parse(tpfsAvroConfig.schema);
+      AvroJob.setInputKeySchema(job, avroSchema);
+      for (Map.Entry<String, String> entry : hConf) {
+        config.put(entry.getKey(), entry.getValue());
+      }
+    } catch (IOException e) {
+      // Shouldn't happen
```

Comment:
Shouldn't happen because ``tpfsAvroConfig.schema`` has been validated before in ``configurePipeline`` method?